CMakeLists.txt

```diff
@@ -251,9 +227,12 @@ add_executable(${testpidmanager_exe} DSI/pid_manager_test.c DSI/pid_manager.c ex
 target_link_libraries(${testpidmanager_exe} ${irods_link_obj_path} "-lm -lglobus_gridftp_server")
 
 set(CMAKE_C_FLAGS "-g -O2 -Wall -lstdc++ -lcurl -O3 -DNDEBUG")
+set(CMAKE_CXX_FLAGS "-g -O2 -Wall -std=c++17 -lstdc++ -lcurl -O3 -DNDEBUG")
```

Comment:
Instead of manually specifying the `-std=c++17` flag, set `CMAKE_CXX_STANDARD`. If we do not want to use `gnu++17` and instead explicitly want `c++17`, use the `CMAKE_CXX_EXTENSIONS` variable.

Additionally, if we are using `libc++`, we probably shouldn't be linking to `libstdc++` too. Are we pulling in a static library that requires it?

Reply:
I tried the part with the CMAKE_CXX_STANDARD a while ago and could never get it to work on this project.  Just tried again and got errors like this:

//usr/include/irods/filesystem/path_traits.hpp:8:16: error: nested namespace definition is a C++17 extension; define each namespace separately [-Werror,-Wc++17-extensions]
namespace irods::experimental::filesystem

Reply:
In that case, I'd say maybe do both for now, and I'll try and figure it out when I do a consistency sweep

Reply:
Got that working with a reordering of CMakeFiles.txt lines.